packages/eslint-plugin-turbo/__tests__/lib/no-undeclared-env-vars.test.ts

```diff
@@ -9,6 +9,7 @@ const ruleTester = new RuleTester({
 const getTestTurboConfig = () => {
   return {
     $schema: "./docs/public/schema.json",
+    globalEnv: ["NEW_STYLE_GLOBAL_ENV_KEY"],
```

Comment:
Let's add a test case where we have an env var that is prefixed with `$` here and in `env` below. 

For example,
```
globalEnv: ["NEW_STYLE_GLOBAL_ENV_KEY", "$NEW_STYLE_GLOBAL_ENV_KEY"],
```

And then this should pass:
```
const val = process.env["$NEW_STYLE_GLOBAL_ENV_KEY"];
```

Reply:
Nice catch. Didn't think of this :)